### PR TITLE
Changes in default value

### PR DESCRIPTION
**Pull request by pajayrao**

### CODE REVIEW

**Comment by ajayraop3495** [comment]
I signed it!

File: tensorflow/python/training/optimizer.py [hunk 1]
```diff
@@ -385,7 +385,7 @@ def _assert_valid_dtypes(self, tensors):
   def _valid_dtypes(self):
     """Valid types for loss, variables and gradients.
 
-    Defaults to `float32`. Subclasses should override to allow other types.
+    Subclasses should override to allow other types.
```
Comment (andrewharp) [inline tensorflow/python/training/optimizer.py]: Would it be more informative but still correct to say "types other than floats"?

File: tensorflow/python/training/optimizer.py [hunk 2]
```diff
@@ -385,7 +385,7 @@ def _assert_valid_dtypes(self, tensors):
   def _valid_dtypes(self):
     """Valid types for loss, variables and gradients.
 
-    Subclasses should override to allow other types.
+    Subclasses should override to allow other float types.
```
Comment (andrewharp) [inline tensorflow/python/training/optimizer.py]: Just to confirm, are float types the only types that could potentially added here? Or are doubles, ints, etc also possible?

File: tensorflow/python/training/optimizer.py (hunk 1 above)
Comment (pajayrao) [inline tensorflow/python/training/optimizer.py]: not really, as loss and gradients values will be float always.

File: tensorflow/python/training/optimizer.py (hunk 2 above)
Comment (andrewharp) [inline tensorflow/python/training/optimizer.py]: Nevermind, just saw your other reply, thanks.
